java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -885,6 +918,28 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams init) {
             );
         }
 
+        private void collectProjectCandidates(FileObject fo, List<FileObject> candidates, AtomicBoolean cancel) {
+            for (FileObject chld : fo.getChildren()) {
+                if (cancel.get()) {
```

Comment:
A weird idea -- consider a symlink in the directory tree; it can loop up the hiererarchy, causing an infinite loop. I'd check somehow absolute paths of the referenced file. Do we want to support symlinks outside of the directory structure ?

Reply:
Fixed - symlinks are excluded from the search.